tx/trigger_WACS_webhooks.py

```diff
@@ -0,0 +1,299 @@
+    if result_json:
+        if 1:
+            print(f"{len(result_json)} user entries received.")
+        else:
+            print("\nUser list:")
+            for j,entry in enumerate(result_json, start=1):
+                print(f"  {j:3}/ {entry['id']:5} '{entry['full_name']:25}' ({entry['username']})")
+        return result_json
+# end of get_all_users()
+
+
+def get_repo_webhook_list(repo_owner_username:str, repo_name:str):
+    """
+    Get the list of webhooks for a given repo.
+    """
+    result_json = get_result(f'repos/{repo_owner_username}/{repo_name}/hooks')
+    if result_json:
+        logging.debug(f"  Received {len(result_json)} webhooks for {repo_owner_username}/{repo_name}")
+        # print(f"\nWebhook list ({len(result_json)}) for repo {repo_owner_username}/{repo_name}:")
+        # for entry in result_json:
+        #     assert entry['config']['content_type'] == 'json'
+        #     print(f"  {entry['id']} {entry['type']}  {entry['config']['url']}  active={entry['active']}")
+        return result_json
+# end of get_repo_webhook_list function
+
+
+def trigger_repo_webhooks(repo_owner_username:str, repo_name:str, webhook_list:List[int]) -> Union[bool,int]:
+    """
+    Trigger the given Gitea webhooks for the given repo.
+
+    Returns the number of webhooks triggered
+        or False if there was an error.
+    """
+    assert repo_owner_username
+    assert repo_name
+    repo_trigger_count = 0
+    logging.info(f"Triggering {repo_owner_username}/{repo_name} webhook{'' if len(webhook_list)==1 else 's'} {webhook_list[0] if len(webhook_list)==1 else webhook_list}…")
+    for webhook_id in webhook_list:
+        result = post(f'repos/{repo_owner_username}/{repo_name}/hooks/{webhook_id}/tests')
+        if result == True:
+            repo_trigger_count += 1
+        else:
+            logging.error(f"Trigger webhook {webhook_id} for {repo_owner_username}/{repo_name} failed!")
+            return False
+    return repo_trigger_count
+# end of trigger_repo_webhooks function
+
+
+def find_and_trigger_repo_webhooks(repo_owner_username:str, repo_name:str) -> bool:
+    """
+    Get a list of webhooks for the given repo
+        and then trigger either the main ones or the DEV- ones or both
+            (depending on the TEST_PREFIXES global setting).
+
+    Returns the number of webhooks triggered
+        or False if there was an error.
+    """
+    assert repo_owner_username
+    assert repo_name
+    logging.info(f"Finding webhooks for {repo_name}/{repo_name}…")
+    repo_webhook_list = get_repo_webhook_list(repo_owner_username, repo_name)
+    if not repo_webhook_list:
+        logging.error(f"No webhooks set for {repo_owner_username}/{repo_name}")
+        return False
+    hook_id_list = []
+    for repo_webhook in repo_webhook_list:
+        logging.debug(repo_webhook)
+        if '' in TEST_PREFIXES \
+        and repo_webhook['config']['url'] in ('https://api.bibletranslationtools.org/webhook'):
```

Comment:
Should be an array of values. Because there is a catalog api too. Unless we don't want to trigger those.